### PR TITLE
BC and DC configuration files

### PR DESCRIPTION
**Pull request by bryan-gilbert**
parameter files for build and deploy for PPR-UI

### CODE REVIEW

File: ppr-ui/openshift/ppr-ui-bc.param [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#=========================================================
+# Template parameters for:
+# Project: ppr-ui
+# Component: build
+# Action: build
```
Comment (WalterMoar) [inline ppr-ui/openshift/ppr-ui-bc.param:5]: What's the difference between component and action?

File: ppr-ui/openshift/ppr-ui-bc.yaml [hunk 2]
```diff
@@ -58,29 +58,24 @@ objects:
...
   displayName: APP Intermediate
   name: APP_INTER_NAME
   required: true
-  value: ppr-ui-inter
 - description: The tag of the intermediate build
   displayName: APP Intermediate Tag
   name: APP_INTER_TAG
   required: true
-  value: latest
 - description: The name of the appliction
```
Comment (WalterMoar) [inline ppr-ui/openshift/ppr-ui-bc.yaml]: Typo in "appliction"

File: ppr-ui/openshift/ppr-ui-bc.param [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# Template File: ppr-ui/openshift/ppr-ui-bc.yaml
+# Hint: oc process -f ppr-ui-bc.yaml --param-file=ppr-ui-bc.param | oc apply -n zwmtib-tools -f -
+#=========================================================
+APP_INTER_NAME="ppr-ui-inter"
+APP_INTER_TAG="latest"
+APP_NAME="ppr-ui"
+APP_PATH="/cooperatives/ppr"
+PPR_GROUP="ppr-ui"
```
Comment (WalterMoar) [inline ppr-ui/openshift/ppr-ui-bc.param:13]: This parameter was removed from the template file.

Comment (WalterMoar) [reply ppr-ui/openshift/ppr-ui-bc.param:13]: Nevermind.

File: ppr-ui/openshift/ppr-ui-dc.dev.param [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#=========================================================
+# Template parameters for:
+# Project: ppr-ui
+# Component: deploy
+# Action: build
```
Comment (WalterMoar) [inline ppr-ui/openshift/ppr-ui-dc.dev.param:5]: Again - what is the difference between component and action? Is the action "build"?

File: ppr-ui/openshift/ppr-ui-dc.prod.param [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+CADDY_CONFIG="ppr-ui-caddy-config"
+ENVIRONMENT="prod"
+IMAGE_TAG="prod"
+LAUNCH_DARKLY_CLIENT_KEY="5ddc04ca5933410824ecf594"
+PAY_API_URL="https://pay-api.pathfinder.gov.bc.ca/api/v1"
+PPR_GROUP="ppr-ui-prod"
+ROUTE_HOST="bcregistry.ca"
+SENTRY_DSN="https://c0eaad46fd3c490882ef1402713c7e6a@sentry.io/1834187"
+SENTRY_ENVIRONMENT="Prod"
+
```
Comment (WalterMoar) [inline ppr-ui/openshift/ppr-ui-dc.prod.param]: GitHub is complaining about the lack of a final newline.

File: ppr-ui/openshift/ppr-ui-dc.test.param [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+CADDY_CONFIG="ppr-ui-caddy-config"
+ENVIRONMENT="test"
+IMAGE_TAG="test"
+LAUNCH_DARKLY_CLIENT_KEY="5ddc04ca5933410824ecf593"
+PAY_API_URL="https://pay-api-test.pathfinder.gov.bc.ca/api/v1"
+PPR_GROUP="ppr-ui-test"
+ROUTE_HOST="test.bcregistry.ca"
+SENTRY_DSN="https://c0eaad46fd3c490882ef1402713c7e6a@sentry.io/1834187"
+SENTRY_ENVIRONMENT="Test"
+
```
Comment (WalterMoar) [inline ppr-ui/openshift/ppr-ui-dc.test.param]: GitHub is complaining about the lack of a final newline.

File: ppr-ui/openshift/ppr-ui-dc.yaml [hunk 7]
```diff
@@ -182,7 +182,7 @@ objects:
         "AUTH_URL": "${AUTH_URL}",
         "LAUNCH_DARKLY_CLIENT_KEY": "${LAUNCH_DARKLY_CLIENT_KEY}",
         "PAY_API_URL": "${PAY_API_URL}",
-        "SENTRY_DSN": "https://${SENTRY_KEY}@sentry.io/1834187"
+        "SENTRY_DSN": "https://${SENTRY_KEY}@sentry.io/1834187",
```
Comment (WalterMoar) [inline ppr-ui/openshift/ppr-ui-dc.yaml]: I don't think we should be committing the 1834187. I think we want to be passing in the entire DSN.

**Comment by bryan-gilbert** [comment]
@WalterMoar both sentry and ld flags have been reset and updated on dev, test and prod.
Have updated the config to use the full sentry url.
Have checked the formatting and the spacing of the dc file is what is desired (so the previous version was not correct).

File: ppr-ui/openshift/ppr-ui-inter-bc.param [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+# Component: deploy
+# Action: build
+# Template File: ppr-ui/openshift/ppr-ui-bc-inter.yaml
+# Hint: oc process -f ppr-ui-inter-bc.yaml --param-file=ppr-ui-inter-bc.param | oc apply -n zwmtib-tools -f -
+#=========================================================
+APP_NAME="ppr-ui-inter"
+APP_PATH="/cooperatives/ppr"
+PPR_GROUP="ppr-ui"
+GIT_URI="https://github.com/bcgov/ppr"
+GIT_REF="master"
```
Comment (WalterMoar) [inline ppr-ui/openshift/ppr-ui-inter-bc.param]: Might as well go alphabetical on this too.

**Review by WalterMoar — CHANGES_REQUESTED** [review]
The `ppr-ui-dc.yaml` file has a lot of changes due to formatting. That extra level of indenting should be removed (oops - just saw comment above... jussec)

**Comment by bryan-gilbert** [comment]
@WalterMoar formatting restored.  For the record the changes came from a activating the Red Hat VSCode YAML extension.

File: ppr-ui/openshift/ppr-ui-dc.yaml [hunk 9]
```diff
@@ -167,44 +167,78 @@ objects:
...
   required: true
-  value: ppr-ui
+- description: The sub-path to the web application
+  displayName: APP path
+  name: APP_PATH
+  required: true
+- description: The full url to the PPR API
+  displayName: API URL
+  name: API_URL
+  required: true
```
Comment (WalterMoar) [inline ppr-ui/openshift/ppr-ui-dc.yaml]: This one is out of order :-|

Comment (bryan-gilbert) [reply ppr-ui/openshift/ppr-ui-dc.yaml]: all in!

**Review by WalterMoar — APPROVED** [review]

**Pull request merged**